File: Python/Easy/717_1-bit_and_2-bit_Characters/solution.py
from typing import List

class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        return Solution.helper(0, bits)
    
    def helper(p, bits):
        if p >= len(bits):
            return False
        if p == len(bits) - 1:
            return True
        
        one = two = False
        if bits[p] == 0:
            one = Solution.helper(p+1, bits)
        elif bits[p] == 1:
            two = Solution.helper(p+2, bits)
        return one or two

File: Golang/Easy/717_1-bit_and_2-bit_Characters/solution.go
package bitand2bitcharacters

func isOneBitCharacter(bits []int) bool {
	return helper(0, bits)
}

func helper(p int, bits []int) bool {
	if p >= len(bits) {
		return false
	}

	if p == len(bits)-1 {
		return true
	}

	one := false
	two := false
	if bits[p] == 0 {
		one = helper(p+1, bits)
	} else if bits[p] == 1 {
		two = helper(p+2, bits)
	}
	return one || two
}
